### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by kieracawley**

### CODE REVIEW

**Comment by kieracawley** [comment]
@rickyc I did it

File: Homework2.html [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  <script>
+    CB.CloudApp.init("hbyttcwntugz", "e4e91c52-fd9d-4ab0-9d44-3659c5ccbf0f");
+
+    class Product{
+      constructor(name, price){
+        this.name = name;
+        this.price = price;
+      }
+      getListItem(){
+        return "<li>" + this.name + ": $" + this.price + "</li>"
```
Comment (rickyc) [inline Homework2.html]: ES6 string interpolation.

File: Homework2.html [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        this.products = [];
+      }
+      listAllProducts(){
+        return this.products;
+      }
+      addProduct(product){
+        this.products.push(product);
+      }
+      updateList(){
+        $("#" + this.name + "List").empty();
```
Comment (rickyc) [inline Homework2.html]: ES6 Strings

**Review by rickyc — CHANGES_REQUESTED** [review]
Close a few more things to solve.

**Comment by rickyc** [comment]
## Homework 2
- [X] Does the application work?
- [X] Is there a HTML / CSS form to create Products?
- [X] Are the Product data saved on the server side (CloudBoost)?
- [X] When I refresh the page, do I see all the products displayed?
- [ ] Can I delete a specific product?
- [X] Is the code written using classes?
- [ ] Does the code make use of conditional selectors?
- [ ] Is there a map that displays all the stores location?
- [X] ~~Does the application make use of getters / setters?~~
- [X] Does the application make use of constructors?
- [X] Does the application properly use const, let and vars?
- [ ] Does the application use proper ES6 syntax? String interpolation, functions.
- [ ] Does it follow the JavaScript code guide conventions?

_Advanced Challenges (Optional)_
- [ ] Are the classes separated into individual files?
- [ ] Is there a Store object on CloudBoost.io?
- [ ] Does the Product / Store object have nested associations?
- [ ] Can the user edit products in the store?
- [ ] Does the code use bootstrap CSS / conventions?

**Comment by rickyc** [comment]
@thebluecat5445 You can merge this one. Make sure you know how to delete products and use conditional selectors. You never addressed those two. 😠

**Comment by kieracawley** [comment]
@rickyc I think I got everything

File: Homework2.html [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+    const stores = {};
+    const storeNames = ["Walmart", "Target", "Staples"];
+
+    for(const name of storeNames) {
+      const store = new Store(name);
+      stores[name] = store;
+    }
+
+    function createNewProduct() {
```
Comment (rickyc) [inline Homework2.html:85]: You don't have to do it for this one, but something to note for the future. You know how there's the `Appdelegate` / `RootViewController` in iOS? You should create a class similar to that to hold all this logic.

Comment (kieracawley) [reply Homework2.html:85]: Ok, thanks for the feedback :)

**Pull request merged**